arch/xtensa/src/common/xtensa_irqdispatch.c

```diff
@@ -94,16 +93,15 @@ uint32_t *xtensa_irq_dispatch(int irq, uint32_t *regs)
 
       /* Update scheduler parameters */
 
-      nxsched_suspend_scheduler(g_running_tasks[this_cpu()]);
-      nxsched_resume_scheduler(this_task());
+      nxsched_suspend_scheduler(*running_task);
```

Comment:
Could you please change all platforms to use *running_task here ? :)

I know at least RISC-V still uses nxsched_suspend_scheduler(g_running_tasks[this_cpu()]);